Source/Session/Session.swift

```diff
@@ -228,6 +242,12 @@ extension Session: VisitableDelegate {
         } else if visitable !== topmostVisit.visitable {
             // Navigating backward
             visit(visitable, action: .restore)
+        } else if isShowingStaleContent {
+            reload()
+            isShowingStaleContent = false
+        } else if isSnapshotCacheStale {
+            clearSnapshotCache()
+            isSnapshotCacheStale = false
```

Comment:
This doesn't seem to work on the `Hiking Journal` demo.
The condition on L242 is different than in https://github.com/hotwired/turbo-ios/pull/173#discussion_r1500859780.
Why are not the flags evaluated separately after the guard statement?

Reply:
You're right, I just confirmed this PR doesn't actually fix the core issue.

I just opened #185 - thanks!